### PR TITLE
[#468] Add clojure.datafy namespace

### PR DESCRIPTION
**Pull request by borkdude**

### CODE REVIEW

**Comment by djblue** [comment]
I can't seem to require `clojure.core.protocols` so I can implement the `Datafiable` protocol.

**Comment by borkdude** [comment]
@djblue I guess that will be part 2 of this exercise.

**Comment by djblue** [comment]
@borkdude it works great otherwise. I was able to datafy `*ns*` and some classes 🎉

**Comment by borkdude** [comment]
@djblue What kinds of objects did you want to implement `Datafiable` protocol for, just out of curiosity?

File: src/babashka/impl/datafy.clj [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+(extend-protocol p/Datafiable
+  SciNamespace
+  (datafy [n]
+    (with-meta {:name (sci-ns-name n)
+                :publics (->> n (sci-ns-publics @ctx) sortmap)
+                :imports (->> n (sci-ns-imports @ctx) sortmap)
+                :interns (->> n (sci-ns-interns @ctx) sortmap)}
+      (meta n))))
+
+(def datafy-ns (create-ns 'clojure.data nil))
```
Comment (cldwalker) [inline src/babashka/impl/datafy.clj:42]: Should this be 'clojure.datafy ?

Comment (borkdude) [reply src/babashka/impl/datafy.clj:42]: Thanks, fixed now.

**Pull request merged**